include/f1x/aasdk/IO/Promise.hpp

```diff
@@ -20,6 +20,7 @@
 
 #include <functional>
 #include <boost/asio.hpp>
+#include <boost/core/noncopyable.hpp>
```

Comment:
thanks for adding this im assuming you also upgraded boost? :p

Reply:
noncopyable is used there, so the header is needed.  Boost from buster somehow includes it implicitly.  buildroot pulls a later version of boost -- and that version, the header isn't implicit and so its needed here.